### PR TITLE
Refactor docs interface pipeline and normalize type display

### PR DESCRIPTION
**Pull request by iMoses**
Do not be afraid of the added lines, it's mostly `package-lock.json` 😅 

Still a WIP but since it's starting to get big we decided to split it to several PRs

### CODE REVIEW

**Comment by iMoses** [comment]
@alantreadway if I installed an external package only for the website should it change the top level `package-lock.json`?

File: packages/ag-charts-website/markdoc.config.ts [hunk 1]
```diff
@@ -86,27 +86,14 @@ export default defineMarkdocConfig({
...
-                interfaceName: { type: String },
-                framework: { type: String },
-                names: { type: String },
-                exclude: { type: String },
-                wrapNamesAt: { type: String },
-                config: { type: String },
+                id: { type: 'String' },
+                include: { type: 'Array' },
+                exclude: { type: 'Array' },
+                noHeader: { type: 'Boolean' },
```
Comment (taktran) [inline packages/ag-charts-website/markdoc.config.ts]: nitpick: I think the convention is `hide`/`show`

```suggestion
                hideHeader: { type: 'Boolean' },
```

Not a big deal though.

File: packages/ag-charts-website/src/features/api-documentation/components/ApiReference.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                if (type.kind === 'union' || type.kind === 'intersection') {
+                    return type.type.map(mapAdditionalType);
+                }
+                console.warn('Unknown type', type);
+            })
+            .filter(Boolean);
+        const codeSample = `function ${interfaceRef.name}(${interfaceRef.type.params
+            .map((param) => `${param.name}: ${normalizeType(param.type)}`)
+            .join(', ')}): ${normalizeType(interfaceRef.type.returnType)};`;
+        // console.log(additionalTypes);
```
Comment (taktran) [inline packages/ag-charts-website/src/features/api-documentation/components/ApiReference.tsx:225]: Can remove this

```suggestion
```

File: packages/ag-charts-community/src/options/chart/chartBuilderOptions.ts [hunk 3]
```diff
@@ -24,3 +29,62 @@ export interface AgChartInstance {
...
+     */
+    download(chart: AgChartInstance, options?: DownloadOptions): void;
+
+    /**
+     * Returns a base64-encoded image data URL for the given `AgChartInstance`.
+     */
+    getImageDataURL(chart: AgChartInstance, options?: ImageDataUrlOptions): Promise<string>;
+}
+
+export interface AgChartSpecialOverrides {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/options/chart/chartBuilderOptions.ts:72]: This shouldn't be public - can we keep it internal, it's mostly for testing and SSR cases?

**Review by alantreadway — COMMENTED** [review]
Looks good on the library side, just one comment.

File: plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/executor-new.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    switch (options.mode) {
+        case 'interfaces':
+            return writeFile(
+                options.output,
+                typeEntries.map(([_, { node, heritage }]) => ({ ...node, heritage }))
+            );
+
+        case 'docs':
+            // return writeFile(options.output, resolveType(typesMap, 'AgBarSeriesOptions'));
```
Comment (taktran) [inline plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/executor-new.ts:40]: Can remove this

```suggestion
```

File: plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/types-utils.ts [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+    let node, heritage;
+    if (typeof typeName === 'object') {
+        node = typeName;
+    } else if (!typesMap.has(typeName)) {
+        console.log('Missing!', typeName);
+        return null;
+    } else {
+        ({ node, heritage } = typesMap.get(typeName));
+    }
+    // console.log({ typeName, node, heritage });
```
Comment (taktran) [inline plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/types-utils.ts:47]: Can remove this

```suggestion
```

File: plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/types-utils.ts [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+                    if (typeName.kind === 'typeLiteral') {
+                        return true;
+                    }
+                    if (typeName.kind === 'typeRef') {
+                        typeName = typeName.type;
+                    }
+                    return !typeName.match(/^['{].*['}]$/);
+                });
+                break;
+            // case 'union':
```
Comment (taktran) [inline plugins/ag-charts-build-tools/src/executors/generate-code-reference-files/types-utils.ts:77]: Not needed?

**Comment by alantreadway** [comment]
> @alantreadway if I installed an external package only for the website should it change the top level `package-lock.json`?

Yes - there is only one `package-lock.json` for the repo, so that is the only place that info can live.

If you check the `node_modules/` folders it should only be installed at the same scope in the repo for which it's installed?

If you need to forcibly split versions, the root `package.json` already has hoisting configuration for the known cases.

File: packages/ag-charts-community/src/options/chart/chartBuilderOptions.ts (hunk 3 above)
Comment (iMoses) [reply packages/ag-charts-community/src/options/chart/chartBuilderOptions.ts:72]: Oh, good to know. I'll fix it 👍🏼

**Review by taktran — APPROVED** [review]
Looks good - `apiReference` looks a lot lot cleaner now ✨ 

Just some minor comments

**Comment by iMoses** [comment]
@taktran I didn't remove some of the comments just because it's a WIP and I'm still using or populating some of these
On the following PR I'll make sure it's all gone, and I also hope I'll have enough time to write all of the correct types so we won't have ts errors on the components

**Pull request merged**